### PR TITLE
Moved complex binding builder implementation to framework. Added method to save browser log

### PR DESCRIPTION
**Pull request by SergeyRuzin**

### CODE REVIEW

File: src/Behavioral.Automation/Services/ComplexBindingBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using NUnit.Framework;
+using TechTalk.SpecFlow;
+
+namespace Behavioral.Automation.Services
+{
+    public class ComplexBindingBuilder : IComplexBindingBuilder
```
Comment (andrey-brovko) [inline src/Behavioral.Automation/Services/ComplexBindingBuilder.cs:10]: Let's add a summary?

File: src/Behavioral.Automation/Services/ComplexBindingBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                RunAction(stepExpression);
+            }
+            TestContext.WriteLine($"{new String(' ', _indendationLevel * _indentationSize)}{stepExpression}");
+        }
+
+        private void RunAction(string stepExpression)
+        {
+            switch (_scenarioContext.StepContext.StepInfo.StepDefinitionType)
+            {
+                case TechTalk.SpecFlow.Bindings.StepDefinitionType.Given:
```
Comment (vladimir-s-quantory) [inline src/Behavioral.Automation/Services/ComplexBindingBuilder.cs:99]: Can we move full type path to `using` directive?

File: src/Behavioral.Automation/Services/ComplexBindingBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        private void BuildAction(object[] pars, IEnumerable<StepDefinitionBaseAttribute> attributes, string methodName)
+        {
+            string definitionType = _scenarioContext.StepContext.StepInfo.StepDefinitionType.ToString();
+            var attributeForCurrentStep = attributes.FirstOrDefault(a => a.GetType().Name.StartsWith(definitionType, StringComparison.OrdinalIgnoreCase));
+            if (attributeForCurrentStep == null)
+            {
+                NUnit.Framework.Assert.Inconclusive($"{definitionType} attribute is not provided for {methodName} binding method");
```
Comment (vladimir-s-quantory) [inline src/Behavioral.Automation/Services/ComplexBindingBuilder.cs:75]: If `methodName` is supposed to be a calling method name you can use this example: 
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=net-6.0#examples

File: src/Behavioral.Automation/Services/ComplexBindingBuilder.cs (hunk 1 above)
Comment (SergeyRuzin) [reply src/Behavioral.Automation/Services/ComplexBindingBuilder.cs:15]: Added

File: src/Behavioral.Automation/Services/ComplexBindingBuilder.cs (hunk 2 above)
Comment (SergeyRuzin) [reply src/Behavioral.Automation/Services/ComplexBindingBuilder.cs]: Moved

File: src/Behavioral.Automation/Services/ComplexBindingBuilder.cs (hunk 3 above)
Comment (SergeyRuzin) [reply src/Behavioral.Automation/Services/ComplexBindingBuilder.cs]: Someone forgot to remove this. Now it's just Assert.Inconclusive

**Review by vladimir-s-quantory — APPROVED** [review]

**Review by vladimir-s-quantory — COMMENTED** [review]
Can we add some kind of release notes here? In some projects we already have ComplexBindings, so it will be handy to know that we have to manually remove them from our projects

File: src/Behavioral.Automation/Services/ComplexBindingBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+using System.Reflection;
+using NUnit.Framework;
+using TechTalk.SpecFlow;
+using TechTalk.SpecFlow.Bindings;
+
+namespace Behavioral.Automation.Services
+{
+    /// <summary>
+    /// Contains BuildAction methods which should be used to call other bindings inside the code (multiple clicks, value selections, etc.)
+    /// This way ctions in called bindings will appear in test logs
```
Comment (andrey-brovko) [inline src/Behavioral.Automation/Services/ComplexBindingBuilder.cs:13]: Typo : 'This way **ctions**'

Comment (SergeyRuzin) [reply src/Behavioral.Automation/Services/ComplexBindingBuilder.cs]: Fixed

**Review by andrey-brovko — APPROVED** [review]

**Review by vladimir-s-quantory — APPROVED** [review]

**Pull request merged**